File: web/tests/reference/stacks/min-stack.js
class MinStack {
  constructor() { this.s = []; this.m = []; }
  push(val) { this.s.push(val); this.m.push(this.m.length ? Math.min(val, this.m[this.m.length - 1]) : val); }
  pop() { this.m.pop(); return this.s.pop(); }
  top() { return this.s[this.s.length - 1]; }
  getMin() { return this.m[this.m.length - 1]; }
}

File: web/tests/reference/stacks/min-stack.py
class MinStack:
    def __init__(self):
        self.s = []
        self.m = []

    def push(self, val):
        self.s.append(val)
        self.m.append(val if not self.m else min(val, self.m[-1]))

    def pop(self):
        self.m.pop()
        return self.s.pop()

    def top(self):
        return self.s[-1]

    def get_min(self):
        return self.m[-1]
